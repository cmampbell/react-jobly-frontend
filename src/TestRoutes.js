import React from 'react'
import { Route, createRoutesFromElements, redirect } from 'react-router-dom'

import JoblyApi from './api'
import { useSearchParams } from './useSearchParams';

import App from './App';
import ErrorBoundary from './ErrorBoundary';

import Home from './Home'
import LoginForm from './LoginForm';
import SignUpForm from './SignUpForm';

import ResultsPage from './ResultsPage'
import CompanyPage from './CompanyPage';
import UserProfile from './UserProfile';

const testCompanies = [
    {
    handle: 'test1',
    name: 'Name1',
    numEmployees: 45,
    description: 'Test company 1',
    jobs: [
        {id: 1, title: 'test1', companyName: 'test', salary: '50000', equity: '.03'},
        {id: 2, title: 'test2', companyName: 'test2', salary: '500', equity: '.05'}]
},
{
    handle: 'test2',
    name: 'Name2',
    numEmployees: 2,
    description: 'Test company 2'
}, 
{
    handle: 'test3',
    name: 'Name3',
    numEmployees: 101,
    description: 'Test company 3'
}, 
]
const testJobs = [
    {id: 1, title: 'test1', companyName: 'test', salary: '50000', equity: '.03'},
    {id: 2, title: 'test2', companyName: 'test2', salary: '500', equity: '.05'}
]

const testUser = {
    username: 'test',
    firstName: 'testFN',
    lastName: 'testLN',
    email: 'test@test.com'
}

const TestRoutes = createRoutesFromElements(
    <>
        <Route element={<App />}
            errorElement={<ErrorBoundary />}
        >
            <Route path='/' element={<Home />} />
            <Route
                path='/companies'
                element={<ResultsPage path='companies' />}
                loader={({ request }) => {
                    return testCompanies
                }
                }
            />
            <Route
                path='/companies/:id'
                element={<CompanyPage />}
                loader={({ params }) => {
                    return testCompanies[params.id]
                }}
            />
            <Route path='/jobs'
                element={<ResultsPage path='jobs' />}
                loader={() => {
                    return testJobs
                }}
            />
            <Route path='/login' element={<LoginForm />} />
            <Route path='/signup' element={<SignUpForm />} />
            <Route
                path='/users/:username'
                element={<UserProfile />}
                loader={async ({ params: { username } }) => {
                    return testUser
                }}
            />
        </Route >
    </>
)

export default TestRoutes;