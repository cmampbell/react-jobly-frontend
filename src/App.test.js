import {render} from '@testing-library/react'
import React from 'react'
import '@testing-library/jest-dom'
import { RouterProvider, createMemoryRouter} from 'react-router-dom'
import { act } from 'react-dom/test-utils';
import TestRoutes from './TestRoutes'
import userEvent from '@testing-library/user-event'
import axios from 'axios';

const router = createMemoryRouter(TestRoutes, {initialEntries: ['/']})
jest.mock('axios')

describe('Smoke and snapshot tests for app', ()=> {
    it('should render without crashing', () => {
        render(<RouterProvider router={router}></RouterProvider>)
      })
    it('should match snapshot', () => {
        const { asFragment } = render(<RouterProvider router={router}></RouterProvider>)
        expect(asFragment()).toMatchSnapshot();
      })
});

describe('Testing register', ()=> {
    it('should take user to signup page when clicking signup link', async ()=> {
        const { getByText, getAllByText, getByPlaceholderText } = render(<RouterProvider router={router}></RouterProvider>)

        await act(()=> userEvent.click(getAllByText('Signup')[0]))
        
        expect(getByText('Username:')).toBeInTheDocument()
        expect(getByText('Password:')).toBeInTheDocument()
        expect(getByText('First name:')).toBeInTheDocument()
        expect(getByText('Last name:')).toBeInTheDocument()
        expect(getByText('Email:')).toBeInTheDocument()

        axios.mockResolvedValue({data: {
            token: 'fakeJWT',
            currentUser: 'testuser'
          }})

        await act(async ()=> {
            userEvent.type(getByPlaceholderText('username'), 'testuser')
            userEvent.type(getByPlaceholderText('password'), 'password')
            userEvent.type(getByPlaceholderText('first name'), 'test')
            userEvent.type(getByPlaceholderText('last name'), 'user')
            userEvent.type(getByPlaceholderText('email'), 'test@user.com')
            await userEvent.click(getByText('Sign Up!'))
        })

        expect(await getByText('Welcome back testuser')).toBeInTheDocument()
    })
})
