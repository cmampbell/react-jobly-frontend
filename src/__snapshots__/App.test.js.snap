// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot tests for app should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="NavBar"
    >
      <a
        aria-current="page"
        class="NavBar-Home active"
        href="/"
      >
        Jobly
      </a>
      <a
        class=""
        href="/login"
      >
        Login
      </a>
      <a
        class=""
        href="/signup"
      >
        Signup
      </a>
    </div>
    <h2>
      You must be a user to see jobs and companies
    </h2>
    <a
      href="/login"
    >
      Login
    </a>
    <a
      href="/signup"
    >
      Signup
    </a>
  </div>
</DocumentFragment>
`;
